### ISSUE TITLE
[Request] How to include fonts in output file

### ISSUE THREAD

**Issue by Hammond95** [gutenbergtools/ebookmaker#106]
I am trying to include font files in the resulting ebook in output.

My project structure is the following:

```
./ebook
├── 00.html
├── 01_copyright_notes.html
├── 02_index.html
├── 03.html
├── 04.html
├── 05.html
[ . . . ]
└── assets
    ├── fonts
    │   ├── CormorantGaramond-Bold.ttf
    │   ├── CormorantGaramond-BoldItalic.ttf
    │   ├── CormorantGaramond-Italic.ttf
    |    [. . .]
    │   └── FabrikCondensed-Bold.ttf
    ├── imgs
    │   ├── cover.jpg
    │   └── wolf.png
    └── styles
        ├── 0.css
        ├── 1.css
        ├── pgepub.css
        └── typography.css
```

I run the following command:
`ebookmaker -v -v --make=epub.images --output-dir=./output/ --max-depth 4 --config ./ebookmaker.conf  ./ebook/00.html`

The output file contains:
```
unzip -l ./output/White\ Fang,\ by\ Jack\ London-images-epub.epub

Archive:  ./output/White Fang, by Jack London-images-epub.epub
  Length      Date    Time    Name
---------  ---------- -----   ----
       20  01-15-2022 07:02   mimetype
      252  01-15-2022 07:02   META-INF/container.xml
   230516  01-15-2022 07:02   OEBPS/3483425176060536139_cover.jpg
    11500  01-15-2022 07:02   OEBPS/3483425176060536139_wolf.png
     1379  01-15-2022 07:02   OEBPS/636569267683567523_0.css
     1738  01-15-2022 07:02   OEBPS/636569267683567523_1.css
      541  01-15-2022 07:02   OEBPS/636569267683567523_pgepub.css
        0  01-15-2022 07:02   OEBPS/636569267683567523_typography.css
      541  01-15-2022 07:02   OEBPS/pgepub.css
      870  01-15-2022 07:02   OEBPS/2457167648625979749_00-0.html.html
     4740  01-15-2022 07:02   OEBPS/2457167648625979749_01_copyright_notes-0.html.html
     5828  01-15-2022 07:02   OEBPS/2457167648625979749_02_index-0.html.html
     2752  01-15-2022 07:02   OEBPS/2457167648625979749_03-0.html.html
    17467  01-15-2022 07:02   OEBPS/2457167648625979749_04-0.html.html
    22686  01-15-2022 07:02   OEBPS/2457167648625979749_05-0.html.html
     8398  01-15-2022 07:02   OEBPS/toc.ncx
      379  01-15-2022 07:02   OEBPS/wrap0000.html
     7756  01-15-2022 07:02   OEBPS/content.opf
---------                     -------
   764129                     45 files
```

The logs WARNINGS are of this kind:
```
Mediatype could not be determined from url file:///Users/mdeluca/_projects/UNIMI-EditoriaDigitale/ebook/assets/fonts/CormorantGaramond-Bold.ttf
```

I see that this log line is produced in this function:

https://github.com/gutenbergtools/ebookmaker/blob/537190734b6994f7e17104a84a862326ca56a708/ebookmaker/Spider.py#L223-L244

but the default appears to be True in case it doesn't recognise the mediatype, so I would expect it to be included.

**Comment by Hammond95** [comment]
I have also noticed that the path in the css files referring to the fonts is being replaced with the one expected in the output epub file.

![image](https://user-images.githubusercontent.com/9408429/149616645-cda6d2f9-1b9c-4632-9e5e-a32cffd5d103.png)

**Comment by eshellman** [comment]
I'll look at it more closely this week, but it may be we need to add a mime type for the fonts file

**Comment by eshellman** [comment]
I think this is easy - can you post a zip for me to test?

**Comment by eshellman** [comment]
the 0.12 branch has fixed this; requires libgutenberg 0.8.14

**Comment by Hammond95** [comment]
Hi @eshellman, now I am getting the following exception:

```python
Traceback (most recent call last):
  File "/Users/mdeluca/.pyenv/versions/3.8.12/lib/python3.8/sre_parse.py", line 1039, in parse_template
    this = chr(ESCAPES[this][1])
KeyError: '\\s'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/mdeluca/.local/share/virtualenvs/UNIMI-EditoriaDigitale-TJ9wuG0W/bin/ebookmaker", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/Users/mdeluca/.local/share/virtualenvs/UNIMI-EditoriaDigitale-TJ9wuG0W/src/ebookmaker/scripts/ebookmaker", line 18, in <module>
    EbookMaker.main ()
  File "/Users/mdeluca/.local/share/virtualenvs/UNIMI-EditoriaDigitale-TJ9wuG0W/src/ebookmaker/ebookmaker/EbookMaker.py", line 574, in main
    dc = get_dc(job) # this is when doc at job.url gets parsed!
  File "/Users/mdeluca/.local/share/virtualenvs/UNIMI-EditoriaDigitale-TJ9wuG0W/src/ebookmaker/ebookmaker/EbookMaker.py", line 232, in get_dc
    dc.add_author('NA', 'cre')
  File "/Users/mdeluca/.local/share/virtualenvs/UNIMI-EditoriaDigitale-TJ9wuG0W/lib/python3.8/site-packages/libgutenberg/DublinCore.py", line 357, in add_author
    name = re.sub(r'\s\s+', r'\s', name)
  File "/Users/mdeluca/.pyenv/versions/3.8.12/lib/python3.8/re.py", line 210, in sub
    return _compile(pattern, flags).sub(repl, string, count)
  File "/Users/mdeluca/.pyenv/versions/3.8.12/lib/python3.8/re.py", line 327, in _subx
    template = _compile_repl(template, pattern)
  File "/Users/mdeluca/.pyenv/versions/3.8.12/lib/python3.8/re.py", line 318, in _compile_repl
    return sre_parse.parse_template(repl, pattern)
  File "/Users/mdeluca/.pyenv/versions/3.8.12/lib/python3.8/sre_parse.py", line 1042, in parse_template
    raise s.error('bad escape %s' % this, len(this))
re.error: bad escape \s at position 0
```

Running `pipenv graph` I have installed

```
- libgutenberg [required: >=0.8.14, installed: 0.8.14]
```

which I found out to be the one on this branch https://github.com/gutenbergtools/libgutenberg/tree/v0_8/libgutenberg

**Comment by Hammond95** [comment]
if I comment this line https://github.com/gutenbergtools/libgutenberg/blob/4da945ed0fccc846aac801bd1d47a6b89ea3a13a/libgutenberg/DublinCore.py#L357 I don't get that exception.

**Comment by Hammond95** [comment]
But apart from that exception, I confirm that now fonts are being correctly included in the output epub.

**Comment by eshellman** [comment]
Lucky I didn't push that to production! tests didn't cover that.
fixed in libgutenberg 0.8.15
